### PR TITLE
test(Accordion): add RTL, add a11y tests, add Accessibility Checker/Axe

### LINKED ISSUE #7204: Overhaul and update Accordion accessibility testing
Definition of done: 
- [x] convert existing tests to React Testing Library
- [x] add Accessibility Checker test
- [x] determine and implement missing WCAG 2.1 tests
- [x] determine and document manual testing practices

### PR DESCRIPTION
**Pull request by dakahn**
Closes #7204  

This PR adds React Testing library and various accessibility tests for Accordion.🏄 

#### Changelog

- adds React Testing Library and a few utilities they provide
- adds automated tests for Axe and AC
- adds keyboard accessibility unit tests
- adds manual testing guide
- adds updated snapshots

#### Testing / Reviewing

run `yarn test Accordion-test.js` and see that there are no errors

### CODE REVIEW

**Review by tw15egan — APPROVED** [review]
No errors on tests, LGTM 👍 ✅

File: packages/react/src/components/Accordion/__tests__/Accordion-test.js [hunk 1]
```diff
@@ -5,16 +5,18 @@
...
  */
 
-import { render, cleanup } from '@carbon/test-utils/react';
-import { mount } from 'enzyme';
 import React from 'react';
 import { default as Accordion, AccordionItem } from '../';
+import { cleanup, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+import { document, it } from 'window-or-global';
```
Comment (joshblack) [inline packages/react/src/components/Accordion/__tests__/Accordion-test.js]:
```suggestion
```

Maybe an auto-import from VSCode?

File: packages/react/src/components/Accordion/__tests__/accordion-avt.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Accordion Component Accessibility Verification Testing
```
Comment (joshblack) [inline packages/react/src/components/Accordion/__tests__/accordion-avt.md:1]: Would it make sense to have this at `components/Accordion/AVT.md`?

**Review by joshblack — APPROVED** [review]
Looks great, left a couple of questions/feedback but should be good to go 🚀

File: packages/react/src/components/Accordion/__tests__/accordion-avt.md (hunk 2 above)
Comment (dakahn) [reply packages/react/src/components/Accordion/__tests__/accordion-avt.md:1]: matched the format of the Link md. Easier for ctrl-p searchin'.

**Pull request merged**